### PR TITLE
Use typescript

### PR DESCRIPTION
**Pull request by greggiannone**
Back to the land of type safety

You can run the backend now with "npm run app" which compiles the typescript into javascript then runs node on the compiled javascript.

I also added linting so the style isn't a complete mess

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,4 +5,4 @@ COPY package*.json ./
 RUN npm install
 COPY . .
```
Comment (OtherwiseJunk) [inline Dockerfile:6]: Can you add a `RUN mkdir /ants` under the copy line here? `copy . .` should in theory copy everything including the ants folder in root, but it wasn't locally and we want to make sure the ants folder is present for the volume mapping to work.

File: src/app.ts [hunk 2]
```diff
@@ -7,13 +7,12 @@ var game = require('./controllers/game');
...
 
 const server = app.listen(port, function() {
-    var host = server.address().address
-    var port = server.address().port
-    console.log("Example app listening at http://%s:%s", host, port)
-})
\ No newline at end of file
+  var host = server.address().address;
+  var port = server.address().port;
+  console.log('Example app listening at http://%s:%s', host, port);
```
Comment (OtherwiseJunk) [inline src/app.ts:17]: could probably update the console log now that it's so close! :P

File: Dockerfile (hunk 1 above)
Comment (greggiannone) [reply Dockerfile:6]: Yep

File: src/app.ts (hunk 2 above)
Comment (greggiannone) [reply src/app.ts:17]: Updated message

**Pull request merged**